extension/data/guiobject/guiobject.js

```diff
@@ -3,22 +3,94 @@ const BYTE_TO_MEGABYTE = 1/1048576;
 const GARBAGE_COLLECTOR_DURATION_WARNING = 100;
 
 
-(function setup() {
-  let elements = [].slice.call(document.querySelectorAll("[data-tooltip]"));
-
-  elements.forEach(function (element) {
-    element.onmouseover = function () {
-      self.postMessage({ type: "update_tooltip", data: this.dataset.tooltip });
-    };
-  });
-
-  // https://bugzilla.mozilla.org/show_bug.cgi?id=660857
-  // TODO: Remove this when bug 660857 is fixed; otherwise
-  // we have to resort to the following workaround:
-  document.getElementById("splash").style.display = "none";
-  document.getElementById("init").style.display = "inline";
+(function init() {
+  function setup(aTarget) {
+    let elements = [].slice.call(document.querySelectorAll("[data-tooltip]"));
+
+    elements.forEach(function (element) {
+      element.onmouseover = function () {
+        aTarget.postMessage({ type: "update_tooltip", data: this.dataset.tooltip }, "*");
+      };
+    });
+
+    // https://bugzilla.mozilla.org/show_bug.cgi?id=660857
+    // TODO: Remove this when bug 660857 is fixed; otherwise
+    // we have to resort to the following workaround:
+    document.getElementById("splash").style.display = "none";
+    document.getElementById("init").style.display = "inline";
+  };
+
+  function australisHelper() {
```

Comment:
Please add at least a description to each of the methods, so it makes it easier for us to understand what those methods are all doing. It will make it also easier to get it to review on AMO.